homeassistant/components/bluetooth/passive_update_coordinator.py

```diff
@@ -157,6 +157,8 @@ def async_add_entities_listener(
         self,
         entity_class: type[PassiveBluetoothCoordinatorEntity],
         async_add_entites: AddEntitiesCallback,
+        entity_filter: Callable[[PassiveBluetoothEntityKey, EntityDescription], bool]
```

Comment:
We could refactor all the existing ones and make this required in a followup

Reply:
Instead I think we should encapsulate the platform entity details within a new class.

- It should hold the entity descriptions, names, entity classes, entity addition callback etc.
- It should have a callback to update the details and create new entities from coordinator data.
- It should be instantiated in the platform and passed to the coordinator in the platform.
- It should be stored by the coordinator.

This way the coordinator doesn't need to know any platform details. The coordinator just let's the platform class know that new data is available and the platform class handles the details. A coordinator can then hold more than one of these platform details instances, and there's a good separation of concerns.

Reply:
I think thats doable.  It might make sense to refactor the existing coordinator to be that new class and then wrap it in the new coordinator class.

I haven't slept yet so will have to chew on that later today after some sleep

Reply:
Basically we split the current coordinator class and put all the entity and platform things in a new base class, which platforms can inherit and/or instantiate as needed.

Reply:
So I think we have the main coordinator class.  And then each platform can register a processor which does the data processing part of what the current one does except coordinating the data.  And the platforms subscribe to their processor which gets data from the coordinator and each has their own update function to process the data

Reply:
Yes. 👍

Reply:
So we have the coordinator which gets the Bluetooth data and hands it to each processor

We have the processor which processes the Bluetooth data and dispatches it to consumers

The consumers are the entities or whatever wants it

Reply:
Yes. The coordinator should return the device data structure to the processor. The processor transforms the device data to the processor representation, eg entities and entity data.

Reply:
I've split it into the Coordinator and Processor (which has an update method per processor) and updated all the integrations.

Its ready for some feedback on the design